src/DataViews/LiveFunctionsDataView.cpp

```diff
@@ -315,13 +311,29 @@ DataView::ActionStatus LiveFunctionsDataView::GetActionStatus(
     return DataView::GetActionStatus(action, clicked_index, selected_indices);
   }
 
-  for (int index : selected_indices) {
-    uint64_t scope_id = GetScopeId(index);
-    const FunctionInfo& instrumented_function = *GetFunctionInfoFromRow(index);
-    if (is_visible_action_enabled(scope_id, instrumented_function)) {
-      return ActionStatus::kVisibleAndEnabled;
-    }
+  std::vector<const FunctionInfo*> function_infos;
+  std::transform(std::begin(selected_indices), std::end(selected_indices),
+                 std::back_inserter(function_infos),
+                 [this](int index) { return GetFunctionInfoFromRow(index); });
+
+  const bool no_functions_selected =
+      all_of(std::begin(function_infos), std::end(function_infos),
+             [](const FunctionInfo* function_info) { return function_info == nullptr; });
+
+  if (no_functions_selected) {
+    return ActionStatus::kInvisible;
   }
```

Comment:
I think we should return `ActionStatus::kVisibleButDisabled` in this case, which means all the actions listed in the above `if` and `else if` cases are greyed out in the context menu.

If we use `kInvisible`, then all those actions are not show up, this might violate the goal of unifying context menu for all views in http://b/205676296. The `kInvisible` is used for the actions that will never be invoked from this view.
And all actions that can be invoked from this view should be visible (either `kVisibleAndEnabled` or `kVisibleButDisabled`).

Reply:
The idea is, all the actions listed here are visible actions. And we let `DataView::GetActionStatus` to determine the status for not listed ones.

Reply:
OK, I see your point. So if the user right-clicks on a manually instrumented scope under Live tab, we should still show hook/unhook and "enable/disable frame track", yet they should be greyed out. 

But wouldn't this be a source of confusion? When I see an entry greyed out, I ask myself "What should I do to make it active?" And there's nothing a user can do to enable these, as the action is either not implemented for manual instrumentation or makes no sense. 

WDYT?

Reply:
The problem is we can select a mixture of dynamic and manually instrumented scopes. The previous unifying was trying to keep the context menu for each view to show consistent entries. But if we short-circuit the context menu when there is only manually instrumented scopes, the visible entries in the context menu will change again depending on the selection.

Reply:
Also, do we consider to implement some of those actions for the manually instrumented scopes in the future?

Reply:
> But if we short-circuit the context menu

Well, I thought, this is the desired behaviour. Anyway, if we want them greyed-out, this only makes the code easier.

>Also, do we consider to implement some of those actions for the manually instrumented scopes in the future?

This is not planned in a foreseenable future. As potentially each manually instrumented scope is actually a user-defined group of manual scopes. So source and disassembly don't make much sense in such a case. Frame track might be implemented.